File: easy/SameTree.java
// https://leetcode.com/problems/same-tree/

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        //Base cases
        if (p == null && q == null)
            return true;
        //if one is null and the other isn't
        if (q == null || p == null)
            return false;
        if (p.val != q.val)
            return false;
            
        //Recursion
        return (isSameTree(p.left, q.left) && isSameTree(p.right, q.right));
        
    }
}

File: easy/same-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */

//  Given the roots of two binary trees p and q, write a
//  function to check if they are the same or not.
//  Two binary trees are considered the same if they are structurally
//  identical, and the nodes have the same value.

 const isSameTree = function(p, q) {
  // recursion
  // base case if p or q is null but not both return false
      // if p and q are null return true

   // check if p.val === q.val, if not return false

  // recursive cases
    // return isSameTree(p.left, q.left) && isSameTree(q.right, q.left)

  if ((!p || !q) && (!p && !q)) return false;

  if (!p && !q) return true;

  if (p.val !== q.val) return false;

  return isSameTree(p.left, q.left) && isSameTree(q.right, q.right);

};
